test/active_model_serializers/model_test.rb

```diff
@@ -18,5 +18,52 @@ def test_initialization_with_string_keys
 
       assert_equal model_instance.read_attribute_for_serialization(:key), value
     end
+
+    def test_attributes_can_be_read_for_serialization
+      klass = Class.new(ActiveModelSerializers::Model) do
+        attributes :one, :two, :three
+      end
+      original_attributes = { one: 1, two: 2, three: 3 }
+      instance = klass.new(original_attributes)
+
+      # Initial value
+      expected_attributes = { id: nil, one: 1, two: 2, three: 3 }.with_indifferent_access
+      assert_equal expected_attributes, instance.attributes
+      assert_equal 1, instance.one
+      assert_equal 1, instance.read_attribute_for_serialization(:one)
+
+      # Change via accessor
+      instance.one = :not_one
+
+      expected_attributes = { id: nil, one: :not_one, two: 2, three: 3 }.with_indifferent_access
+      assert_equal expected_attributes, instance.attributes
+      assert_equal :not_one, instance.one
+      assert_equal :not_one, instance.read_attribute_for_serialization(:one)
+    end
```

Comment:
is fixed https://github.com/rails-api/active_model_serializers/pull/1903 

ref https://github.com/rails-api/active_model_serializers/issues/1877